core/src/test/java/io/aiven/klaw/service/UsersTeamsControllerServiceTest.java

```diff
@@ -239,7 +237,86 @@ public void resetPasswordGenerateToken_withFailure() throws KlawException {
+    when(mailService.getUserName(any())).thenReturn("testUserName");
+    when(commonUtilsService.getTenantId(anyString())).thenReturn(101);
+    when(handleDbRequests.getTeamDetails(101, 101)).thenReturn(teamDaoMock);
+
+    TeamModelResponse response = usersTeamsControllerService.getTeamDetails(101, "testTenantName");
+
+    assertThat(response.getTeamname()).isEqualTo(teamDaoMock.getTeamname());
+    assertThat(response.getTeamphone()).isEqualTo(teamDaoMock.getTeamphone());
+    assertThat(response.getContactperson()).isEqualTo(teamDaoMock.getContactperson());
+    assertThat(response.getTeamId()).isEqualTo(teamDaoMock.getTeamId());
+    assertThat(response.getTenantId()).isEqualTo(tenantMapMock.entrySet().stream()
+            .filter(obj -> Objects.equals(obj.getValue(), "testTenantName"))
+            .findFirst()
+            .get()
+            .getKey());
+    assertThat(response.getTenantName()).isEqualTo("testTenantName");
+    assertThat(response.getTeammail()).isEqualTo(teamDaoMock.getTeammail());
+    assertThat(response.getApp()).isEqualTo(teamDaoMock.getApp());
+    assertThat(response.getEnvList()).isEqualTo(List.of(teamDaoMock.getRequestTopicsEnvs().split("\\s*,\\s*")));
+    assertThat(response.getServiceAccounts()).isEqualTo(teamDaoMock.getServiceAccounts());
+  }
+
+  @Test
+  void getTeamDetailsNoRequestTopicsEnvs() {
+    Team teamDaoMock = utilMethods.getTeamDaoMock();
+    teamDaoMock.setRequestTopicsEnvs(null);
+    Map<Integer, String> tenantMapMock = utilMethods.getTenantMapMock();
+
+    when(manageDatabase.getTenantMap()).thenReturn(tenantMapMock);
+    when(mailService.getUserName(any())).thenReturn("testUserName");
+    when(commonUtilsService.getTenantId(anyString())).thenReturn(101);
+    when(handleDbRequests.getTeamDetails(101, 101)).thenReturn(teamDaoMock);
+
+    TeamModelResponse response = usersTeamsControllerService.getTeamDetails(101, "testTenantName");
+
+    assertThat(response.getTeamname()).isEqualTo(teamDaoMock.getTeamname());
+    assertThat(response.getTeamphone()).isEqualTo(teamDaoMock.getTeamphone());
+    assertThat(response.getContactperson()).isEqualTo(teamDaoMock.getContactperson());
+    assertThat(response.getTeamId()).isEqualTo(teamDaoMock.getTeamId());
+    assertThat(response.getTenantId()).isEqualTo(tenantMapMock.entrySet().stream()
+            .filter(obj -> Objects.equals(obj.getValue(), "testTenantName"))
+            .findFirst()
+            .get()
+            .getKey());
+    assertThat(response.getTenantName()).isEqualTo("testTenantName");
+    assertThat(response.getTeammail()).isEqualTo(teamDaoMock.getTeammail());
+    assertThat(response.getApp()).isEqualTo(teamDaoMock.getApp());
+    assertThat(response.getEnvList()).isNull();
+    assertThat(response.getServiceAccounts()).isEqualTo(teamDaoMock.getServiceAccounts());
+  }
+
+  @Test
+  void getTeamDetailsTeamDoesNotExist() {
+    when(mailService.getUserName(any())).thenReturn("testUserName");
```

Comment:
We should define the test user details as method or class variables as well, tenantId and teamId can probably be set at the class level but the username and team name can probably be method level.